Misc/NEWS.d/next/Library/2025-06-08-11-11-07.gh-issue-135234.wJCdh0.rst

```diff
@@ -0,0 +1,2 @@
+:mod:`hashlib`: improve exception messages when an OpenSSL function failed.
+Patch by Bénédikt Tran.
```

Comment:
```suggestion
:mod:`hashlib`: improve exception messages when an OpenSSL function failed.
When allocation fails on OpenSSL's side, a :exc:`MemoryError` is raised
instead of a :exc:`ValueError`. Patch by Bénédikt Tran.
```